calibration_residual_kernals.py

```diff
@@ -136,29 +139,27 @@ def calc_residuals_from_ifft(ant_to_gains, observation_frequencies, current_phas
                     stream_idx = int(str(tune)+str(pol),2)
                     
                     #Calculating the power here and SNR
-                    #signal_pow = ifft_abs_matrix[stream_idx, max_idxs[stream_idx]]
-                    #SNR[antenna_index, stream_idex] = (signal_pow - median[stream_idx])/sigma[stream_idx]
+                    snr[stream_idx] = (ifft_abs_matrix[stream_idx, max_idxs[stream_idx]] - ant_median[stream_idx])/ant_sigma[stream_idx]
 
                     residual_delay = -1.0 * tlags[max_idxs[stream_idx]]
                     #Now rather than using the abs(residual_delay) value, use
-                    #if SNR[antenna_index, stream_idex] > 4.0, then update the delay values, if not there is no delay peak and probably np.argmax will 
-                    #pick up some random noise and no point in updating the delay values
-                    
-                    if np.abs(residual_delay) > delay_residual_rejection:
-                        residual_delays[stream_idx] = 0.0
-                        phase_cals[stream_idx] = phase_matrix[stream_idx]
-                    else:
+                    if snr[stream_idx] > snr_threshold:
                         residual_delays[stream_idx] = residual_delay
                         gain_from_residual_delay = np.exp(2j*np.pi*(observation_frequencies[tune,:]*1e-9)*residual_delays[stream_idx])
                         phase_cals[stream_idx] = np.angle(new_gain_matrix[stream_idx,:]/gain_from_residual_delay)
                         #zero all phases outside the collected gains range
                         phase_cals[stream_idx, uncollected_gain_range] = 0.0
+                    else:
+                        residual_delays[stream_idx] = 0.0
+                        phase_cals[stream_idx] = phase_matrix[stream_idx]
                     
                     #Calculating the spread of phases collected from each antennas from regions where we have actual gain values
-                    #sigma_phase[antenna_index, stream_idex] = np.std(phase_cals[stream_idx, collected_gain_range], axis = 1)
+                    sigma_phase[stream_idx] = np.std(phase_cals[stream_idx, frequency_indices[tune]], axis = 1)
 
             delay_residual_map[ant] = residual_delays
             phase_cal_map[ant] = phase_cals
+            snr_map[ant] = snr
```

Comment:
Do we want to return these dictionaries of `{ant : [snr]}` and `{ant : [sigma_phase]}` from the kernel for plotting in the slack channel? Is this a useful metric for visualisation at all - obviously outside of being a part of the rejection criteria...

Reply:
I think this is definitely a useful metric to print out somewhere in slack.